### ISSUE TITLE
simulator bug: car remained stationary

### ISSUE THREAD

**Issue by asimonov** [Kairos-Automotive/carla-brain#106]
The car remained stationary for a long time even though it was supposed to move.

**Comment by yhoazk** [comment]
Maybe related to #62

**Comment by asimonov** [comment]
could be.

@Kairos-Automotive/udacity , i wonder if this is just a resource constraint. does anyone reproduce it? for me it works, even on laptop

**Comment by yhoazk** [comment]
Yes, an issue due to resource constraint. I'm able to reproduce it. I'm trying to find where is the delay, because the detector, classifier and updater finish in time (apparently). But the car does not move.

**Comment by asimonov** [comment]
Oh, this is great!

_____________________________
Kind Regards,
Alexey Simonov

**Comment by yhoazk** [comment]
The most probable root cause of this issue is a resource starvation. 
In my environment the websocket randomly stops the transmission of `steering`, `brake` and `throttle`.
The callbacks are being called, but the messages are not arriving to the simulator. I was not able to find any error log in the logs, even when activating the debug option `server.py:12::sio = socketio.Server(engineio_logger=True)`. 
@avrabe  Do you know some method to verify that the socketio Tx connections is still alive?

**Comment by asimonov** [comment]
Good we could get closer to proving this!

There is not much we can do about the reviewer's spec, apart from mentioning it in submission. But it looks like they ignore the commentary.

Alternatively we can try to squeeze more performance out of the whole solution, but this is not trivial.

I suggest we leave this issue unfixed for next submission. Hopefully it goes to a reviewer with better hardware and there will be no issue.

**Comment by asimonov** [comment]
BTW, can you reproduce the issues even when switching from real detector to ground truth data?
